kolibri/plugins/device/assets/src/views/DeviceManagementSideNavEntry.vue

```diff
@@ -1,36 +1,166 @@
     },
+    mixins: [commonCoreStrings],
+    data() {
+      return {
+        visibleSubMenu: false,
+        isSubsetOfUsersDevice: plugin_data.isSubsetOfUsersDevice,
+      };
+    },
+    mounted() {
+      this.submenuShouldBeOpen();
+    },
+    computed: {
+      ...mapGetters(['isAppContext']),
+      url() {
+        return urls['kolibri:kolibri.plugins.device:device_management']();
+      },
+      routes() {
+        return {
+          channels: {
+            text: this.coreString('channelsLabel'),
+            route: this.generateNavRoute(DevicePageNames.MANAGE_CONTENT_PAGE),
+          },
+        };
+      },
+      generalDeviceRoutes() {
+        return {
+          permissions: {
+            text: this.$tr('permissionsLabel'),
+            route: this.generateNavRoute(DevicePageNames.MANAGE_PERMISSIONS_PAGE),
+          },
+          facilities: {
+            text: this.coreString('facilitiesLabel'),
+            route: this.generateNavRoute(DevicePageNames.FACILITIES_PAGE),
+          },
+          info: {
+            text: this.$tr('infoLabel'),
+            route: this.generateNavRoute(DevicePageNames.DEVICE_INFO_PAGE),
+          },
+          settings: {
+            text: this.$tr('settingsLabel'),
+            route: this.generateNavRoute(DevicePageNames.DEVICE_SETTINGS_PAGE),
+          },
+        };
+      },
+      iconAfter() {
+        if (this.isAppContext) {
+          return this.visibleSubMenu ? 'chevronUp' : 'chevronDown';
+        }
+      },
+    },
+    methods: {
+      generateNavRoute(route) {
+        return generateNavRoute(this.url, route, baseRoutes);
+      },
+      toggleAndroidMenu() {
+        this.$emit('toggleAndroidMenu');
+      },
+      isActiveLink(route) {
+        return route.includes(this.$router.currentRoute.path);
+      },
+      submenuShouldBeOpen() {
+        // which plugin are we currently in?
+        this.visibleSubMenu = window.location.pathname.includes(this.url);
+      },
+      subpathStyles(route) {
+        if (this.isActiveLink(route)) {
+          console.log('active');
+          return {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+            textDecoration: 'none',
+          };
+        }
+        return {
+          color: this.$themeTokens.text,
+          textDecoration: 'none',
+          ':hover': {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+          },
+          ':focus': this.$coreOutline,
+        };
+      },
+      handleNav(route) {
+        this.isActiveLink(route) ? this.toggleAndroidMenu() : null;
+      },
+    },
     $trs: {
       device: {
         message: 'Device',
         context:
           'The device is the physical or virtual machine that has the Kolibri server installed on it.',
       },
-    },
-    computed: {
-      url() {
-        return urls['kolibri:kolibri.plugins.device:device_management']();
+      permissionsLabel: {
```

Comment:
Can move these into CommonDeviceStrings to avoid duplication with https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/views/DeviceTopNav.vue#L68

Reply:
These strings appear to be new here still.